### PR TITLE
refactor: cleanup colors

### PR DESCRIPTION
**Pull request by markov00**
## Summary

During the heatmap legend fixes I've noticed that we have multiple places where some standard colors are allocated.
I've moved the `Color` type + some static colors (red, white, black, transparent) into the `common/color.ts` file.
This is not a complete refactoring but the first part of the initiative to cleanup the folders the typings.

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)

### CODE REVIEW

File: packages/charts/src/common/color.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export const WHITE_COLOR: Color = 'rgba(255, 255, 255, 1)';
+/** @internal */
+export const BLACK_COLOR: Color = 'rgba(0, 0, 0, 1)';
+/** @internal */
+export const RED_COLOR: Color = 'rgba(255, 0, 0, 1)';
+
+/** @internal */
+export const RED_RGBOBJ_COLOR: RgbObject = { r: 255, g: 0, b: 0, opacity: 1 };
+/** @internal */
+export const TRANSPARENT_RBGOBJ_COLOR: RgbObject = { r: 0, g: 0, b: 0, opacity: 0 };
```
Comment (nickofthyme) [inline packages/charts/src/common/color.ts:26]: What would you think about putting these in an `object` or `class` a la Flutter (https://api.flutter.dev/flutter/material/Colors-class.html). That way we lessen the import mess and the less redundant naming, `RED_COLOR` would just be `Colors.Red`.

That way we still preserve the `Color` as a type.

File: storybook/stories/goal/10_band_in_band.tsx [hunk 2]
```diff
@@ -12,8 +12,8 @@ import { Chart, Goal, Settings } from '@elastic/charts';
 import { config } from '@elastic/charts/src/chart_types/goal_chart/layout/config/config';
 import { BandFillColorAccessorInput } from '@elastic/charts/src/chart_types/goal_chart/specs';
 import { GoalSubtype } from '@elastic/charts/src/chart_types/goal_chart/specs/constants';
-import { Color } from '@elastic/charts/src/utils/common';
 
+import { Color } from '../../../packages/charts/src/common/color';
 import { useBaseTheme } from '../../use_base_theme';
 
 const subtype = GoalSubtype.Goal;
```
Comment (nickofthyme) [inline storybook/stories/goal/10_band_in_band.tsx:19]: ??? Why is this changed? I think your IDE is playing tricks on you.

**Review by nickofthyme — COMMENTED** [review]
LGTM, I would suggest a slightest simplified API. See comments.

File: packages/charts/src/common/color.ts (hunk 1 above)
Comment (nickofthyme) [reply packages/charts/src/common/color.ts:26]: You could even do `Colors.rgbo.Red` or something.

**Review by monfera — APPROVED** [review]
LGTM, good to merge once it passes and conflicts are resolved

Switching to some color object (or fake enum) would be sensible if we expect significant growth to the count of these constants, or a `COLOR_` prefix at least. With the current handful of colors, I don't feel strongly either way.  Worth doing a last minute grep for all string occurrences `'black'` or even without quotes to ferret out template string occurrences or similar

File: packages/charts/src/common/color.ts [hunk 3]
```diff
@@ -12,15 +12,21 @@ import { RgbObject } from './color_library_wrappers';
...
+  },
+  Black: {
+    rgba: 'rgba(0, 0, 0, 1)',
+    rgbaChannels: { r: 0, g: 0, b: 0, opacity: 1 },
+  },
+  Transparent: {
+    rgba: 'rgba(0, 0, 0, 0)',
+    rgbaChannels: { r: 0, g: 0, b: 0, opacity: 0 },
+  },
+};
```
Comment (nickofthyme) [inline packages/charts/src/common/color.ts:32]: ❤️ Nice! The merging of like colors was an added perk! 👍🏼

**Review by nickofthyme — APPROVED** [review]

**Comment by markov00** [comment]
just an update: I've found that `transparent` and `rgba(0,0,0,0)` are not treated the same way in the partition chart. I'm currently investigating to fix this

**Comment by markov00** [comment]
close in favour of #1397
